pandas/core/generic.py

```diff
@@ -188,6 +190,9 @@ class NDFrame(PandasObject, SelectionMixin):
     _is_copy = None
     _data = None  # type: BlockManager
 
+    if TYPE_CHECKING:
```

Comment:
Why is this required in the TYPE_CHECKING block?

Reply:
Otherwise you'd have a mutable default shared by all the classes. Not sure if there's a better way, but this is the workaround I saw on some mypy thread.

Reply:
Hmm can you maybe share the error? We already have the `_metadata` defaulting to `[]` so not clear on the distinction.

Reply:
There wouldn't be an error, it'd typecheck just fine.

It's a runtime issue though, since you could do `pd.Series.attrs['a'] = 'foo'`, which would affect all the instances.

With python 3.6+, we can just do

```
attrs : Dict[Hashable, Any]
```

but the workaround for 3.5 is to put it in a TYPE_CHECKING block.

Reply:
Ah OK. I wasn't getting the link between runtime and TYPE_CHECKING but understood now. So I guess should either wait for Py35 to drop or should add a TODO here to clean up when that happens and add the definition without declaration